### ISSUE TITLE
Add CORS headers to API

### ISSUE THREAD

**Issue by fabrixxm** [friendica/friendica#1214]
To ease life for html-based 3th part applications

https://en.wikipedia.org/wiki/Cross-origin_resource_sharing

**Comment by Rudloff** [comment]
I would like to work on a PR for this.
What header should we add? `Access-Control-Allow-Origin: *` or something more restrictive?

**Comment by MrPetovan** [comment]
We don’t support multi-domains so it should be the server base URL.

**Comment by Rudloff** [comment]
This would mean that no web app on another domain could access the API.
If we do this, we should probably have a way to authorize other domains.

**Comment by MrPetovan** [comment]
I don’t know any web app that could work with Friendica’s API, not even our own web app is using the API, but just in case it should be able to do so.

If needs arise, we can expand the header to include either a hard-coded list of authorized web apps or a configurable list of domains.

**Comment by Rudloff** [comment]
It's not really useful to set the `Access-Control-Allow-Origin` header with only the server's own domain, since this is already the default behavior when the header is not set.
So I guess there's nothing to do for now.

**Comment by tobiasd** [comment]
> I don’t know any web app that could work with Friendica’s API, not
> even our own web app is using the API, but just in case it should be
> able to do so.

What about the GNUsocial apps (e.g. Quitter). If it is using the API it should work with Friendica as well.

**Comment by MrPetovan** [comment]
As far as I could find out about “Quitter” or rather “Qvitter”, this is just a UI theme for GNU Social instances to choose from, but not a complete web app that would be able to connect to any GNU Social-capable server.

**Comment by rabuzarus** [comment]
Yes, quitter is a theme for gnu social.  But it is getting all its data through the api.

**Comment by tobiasd** [comment]
> Yes, quitter is a theme for gnu social.  But it is getting all its
> data through the api.

I see, I had the impression it was a stand-alone app, that gets the data via API. Thanks for clarification!

**Comment by fabrixxm** [comment]
+1 for a way to authorize domains

API users will be mostly users of the instance, so could be something users can request to admins
to approve...

**Comment by nupplaphil** [comment]
Fixed per #10428
